tests/jballerina-unit-test/src/test/resources/test-src/statements/ifelse/test_type_guard_type_narrow_1.bal

```diff
@@ -0,0 +1,557 @@
+// Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+function test1(int|string s) {
+    if s is int {
+        int _ = s;
+    } else {
+        string _ = s;
+    }
+
+    int _ = s; // error incompatible types: expected 'int', found '(int|string)'
+}
+
+function test2(int|string|boolean b) {
+    if b is string {
+        string _ = b;
+    } else if b is boolean {
+        boolean _ = b;
+    } else {
+        int _ = b;
+    }
+
+    int _ = b; // error incompatible types: expected 'int', found '(int|string|boolean)'
+}
+
+function test3(int|string|boolean? b) {
+    if b is string {
+        string _ = b;
+    } else if b is boolean {
+        boolean _ = b;
+    } else {
+        int? _ = b;
+    }
+
+    int _ = b; // error incompatible types: expected 'int', found '(int|string|boolean)?'
+}
+
+function test4(int|string s) {
+    if s is int {
+        int _ = s;
+        return;
+    }
+
+    string _ = s;
+}
+
+function test5(int|string|boolean b) {
+    if b is string {
+        string _ = b;
+        return;
+    } else if b is boolean {
+        boolean _ = b;
+        return;
+    }
+
+    int _ = b; // type not narrowed. issue #34307
+}
+
+function test6(int|string|boolean? b) {
+    if b is string? {
+        string? _ = b;
+        return;
+    } else if b is boolean {
+        boolean _ = b;
+        return;
+    }
+
+    int _ = b; // type not narrowed. issue #34307
+}
+
+function test7(boolean? b) {
+    if b is () {
+        () _ = b;
+        return;
+    }
+
+    boolean _ = b;
+}
+
+function test8(boolean? b) {
+    if b is () {
+        () _ = b;
+        return;
+    } else {
+        boolean _ = b;
+    }
+
+    boolean _ = b; // type not narrowed. issue #34307
+}
+
+function test9(boolean? b) {
+    if b is true {
+        true _ = b;
+        return;
+    } else if b is false {
+        false _ = b;
+        return;
+    } else {
+        () _ = b; // type not narrowed. issue #30598, #33217
+    }
+
+    () _ = b; // type not narrowed. issue #34307
+}
+
+function test10(true|false? b) {
```

Comment:
Let's add tests using !is and logical exp as well if not already covered
```ballerina
function foo(int|string r) {
    if r is int || r !is string {
        int _ = r;  //error incompatible types: expected 'int', found '(int|string)'
    } 
}
```

Reply:
We had these covered in `test_type_guard_type_narrow_3.bal`. Anyway, I had written them using the `!(r is string)` form. Duplicated those tests and included `r !is string` form also.